File: msa-with-spring-cloud/user-service/src/main/java/dev/yoon/userservice/repository/UserRepository.java
package dev.yoon.userservice.repository;

import dev.yoon.userservice.entity.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<UserEntity, Long> {

    Optional<UserEntity> findByUserId(String userId);

    Optional<UserEntity> findByEmail(String email);
}

File: developer's-note-of-incorrect-answer-for-adding-tests/kotlin-test-code-with-architecture/src/main/kotlin/com/example/kotlintestcodewitharchitecture/repository/UserRepository.kt
package com.example.kotlintestcodewitharchitecture.repository

import com.example.kotlintestcodewitharchitecture.model.UserStatus
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<UserEntity, Long> {

    fun findByIdAndStatus(id: Long, userStatus: UserStatus) : UserEntity?
    fun findByEmailAndStatus(email: String, userStatus: UserStatus) : UserEntity?
}